src/guide/3-behaviour-definition/1-defining-functions.rst

```diff
@@ -93,6 +93,33 @@ The function can then be added to an ``AgentDescription`` using the ``newRTCFunc
     // The AgentFunctionDescription is stored in the agent_fn1_description variable
     AgentFunctionDescription& agent_fn1_description = agent.newRTCFunction("agent_fn1", agent_fn1_source);
 
+FLAMEGPU Device Functions
+-------------------------
+
+If you wish to define regular functions which can be used within agent function definitions, you can use the `FLAMEGPU_DEVICE_FUNCTION` macro:
+
+.. tabs::
+
+  .. code-tab:: cpp
+
+    // Define a function for adding two ingtegers which can be called inside agent functions.
+    FLAMEGPU_DEVICE_FUNCTION int add(int a, int b) {
+      return a + b;
+    }
+
+FLAMEGPU Host Device Functions
+------------------------------
+
+If you wish to define regular functions which can be used within agent function definitions and in host code, you can use the `FLAMEGPU_HOST_DEVICE_FUNCTION` macro:
```

Comment:
Probably need a note about how this won't really work for RTC, because RTC is device only compilation etc etc.

Reply:
Also perhaps "called from" rather than "used within".